### PR TITLE
S138 doc improvements

### PR DESCRIPTION
**Pull request by eschultink**
Direct to main, as docs pertain to all versions > 0.4.8.

### Features
  - add more guidance for how Terraform State backends should be configured

### Change implications

 - dependencies added/changed? **no**

### CODE REVIEW

File: docs/gcp/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Getting Started - GCP
+
+## Prerequisites
+
+A Google (GCP) user who can create projects (or with sufficient permissions to provision Service 
+Accounts, Keys, GCS buckets, and Cloud Functions within an existing project).
```
Comment (aperez-worklytics) [inline docs/gcp/getting-started.md:6]:
```suggestion
Accounts, Keys, Secret Manager, GCS buckets, and Cloud Functions within an existing project).
```

File: infra/examples/aws-google-workspace/main.tf [hunk 2]
```diff
@@ -14,9 +14,22 @@ terraform {
     }
   }
 
-  # if you leave this as local, you should backup/commit your TF state files
+  # we recommend you use a secure location for your Terraform state (such as S3 bucket), as it
+  # may contain sensitive values (such as API keys) depending on which data sources you configure.
+  #
+  # local may be safe for production-use IFF you are executing Terraform from a secure location
```
Comment (aperez-worklytics) [inline infra/examples/aws-google-workspace/main.tf:20]:
```suggestion
  # local may be safe for production-use IF you are executing Terraform from a secure location
```

File: infra/examples/aws-msft-365/main.tf [hunk 3]
```diff
@@ -12,9 +12,22 @@ terraform {
     }
   }
 
-  # if you leave this as local, you should backup/commit your TF state files
+  # we recommend you use a secure location for your Terraform state (such as S3 bucket), as it
+  # may contain sensitive values (such as API keys) depending on which data sources you configure.
+  #
+  # local may be safe for production-use IFF you are executing Terraform from a secure location
```
Comment (aperez-worklytics) [inline infra/examples/aws-msft-365/main.tf:18]:
```suggestion
  # local may be safe for production-use IF you are executing Terraform from a secure location
```

File: infra/examples/gcp-google-workspace/main.tf [hunk 4]
```diff
@@ -5,9 +5,21 @@ terraform {
     }
   }
 
-  # if you leave this as local, you should backup/commit your TF state files
+  # we recommend you use a secure location for your Terraform state (such as GCS bucket), as it
+  # may contain sensitive values (such as API keys) depending on which data sources you configure.
+  #
+  # local may be safe for production-use IFF you are executing Terraform from a secure location
```
Comment (aperez-worklytics) [inline infra/examples/gcp-google-workspace/main.tf:11]:
```suggestion
  # local may be safe for production-use IF you are executing Terraform from a secure location
```

File: java/core/core.iml [hunk 5]
```diff
@@ -10,7 +10,6 @@
       <sourceFolder url="file://$MODULE_DIR$/src/test/resources" type="java-test-resource" />
       <sourceFolder url="file://$MODULE_DIR$/target/generated-sources/annotations" isTestSource="false" generated="true" />
       <sourceFolder url="file://$MODULE_DIR$/target/generated-test-sources/test-annotations" isTestSource="true" generated="true" />
-      <excludeFolder url="file://$MODULE_DIR$/target" />
```
Comment (aperez-worklytics) [inline java/core/core.iml:13]: Now included? (Not sure how it is in latest version)

**Review by aperez-worklytics — APPROVED** [review]

File: infra/examples/aws-msft-365/main.tf (hunk 3 above)
Comment (eschultink) [reply infra/examples/aws-msft-365/main.tf:18]: sry, `IFF` is math notation for "if and only if"

**Pull request merged**